stdlib/asm/mem.masm

```diff
@@ -0,0 +1,51 @@
+# ===== MEMORY FUNCTIONS ==========================================================================
+
+#! Copies `n` words from `read_ptr` to `write_ptr`.
+#!
+#! Stack transition looks as follows:
+#! [n, read_ptr, write_ptr, ...] -> [...]
+#! cycles: 15 + 16n
+export.memcopy
+  # The loop variable is changed with an add instead of sub because the former
+  # uses one fewer cycle. So here the counter is negated. (1 cycles)
+  # stack: [-n, read_ptr, write_ptr, ...]
+  neg
+
+  # Pad the stack because mem_load overwrites it (4 cycles)
+  # stack: [0, 0, 0, 0, -n, read_ptr, write_ptr, ...]
+  padw
+
+  # check loop condition (3 cycles)
+  # stack: [b, 0, 0, 0, 0, -n, read_ptr, write_ptr, ...]
+  dup.4 neq.0
+
+  # LOOP: [0, 0, 0, 0, -n, read_ptr, write_ptr, ...]
+  # while(n!=0) (16 cycles)
+  while.true
```

Comment:
I'm getting 19 cycles rather than 16.

Reply:
(sorry I missed this last time)

Reply:
Are you counting `add.1` as one or two cycles? (it reduces to `INCR` operation which takes one cycle).

Reply:
Ahh good point, thank you! Yes, my mistake. All looks good!